### PR TITLE
Fix/85 nav expose

### PR DESCRIPTION
**Pull request by cbkerr**
## Description

Apply reorganization suggestions from #85.
I left FAQ under Topic Guides because it didn't feel like a "getting started" section.

## Checklist:

- [x] I am familiar with the [**Contributing Guidelines**](https://github.com/glotzerlab/signac-docs/blob/master/CONTRIBUTING.md).
- [x] I agree with the terms of the [**Contributor Agreement**](https://github.com/glotzerlab/signac-docs/blob/master/ContributorAgreement.md).
- [ ] My code follows the [code style guideline](https://github.com/glotzerlab/signac-docs/blob/master/CONTRIBUTING.md#code-style) of this project.

### CODE REVIEW

**Comment by cbkerr** [comment]
I think the changes in this PR make #96 make more sense.

File: docs/source/packages.rst [hunk 1]
```diff
@@ -1,15 +1,9 @@
 .. _package-overview:
 
 ==============
-Packages (API)
+Packages
 ==============
```
Comment (csadorf) [inline docs/source/packages.rst:5]: Make sure to adjust the length of the over- and underlines as well.

File: docs/source/recipes.rst [hunk 2]
```diff
@@ -1,7 +1,7 @@
 .. _recipes:
 
 =======
-Recipes
+Advanced How-To
```
Comment (csadorf) [inline docs/source/recipes.rst:4]: see above

**Comment by cbkerr** [comment]
> Is this built anywhere to check out?

I'll try to set that up.

**Comment by csadorf** [comment]
The docs for this branch are now built here: https://docs.signac.io/en/fix-85-nav-expose/

**Comment by cbkerr** [comment]
For the sake of completeness, I'm adding the previously left-out sections to the sidebar to start a conversation on how many elements should be revealed in the sidebar.

One solution could be to divide "Topic Guide" into a few more headings, mirroring the groupings used previously: "Managing Data", "Implementing Workflows", and "Additional Topics".

**Comment by cbkerr** [comment]
> I think I'd prefer the Package API Reference to be above Topic Guide, just so that all 3 main headings are viewable without scrolling.

I agree!

**Pull request merged**